### ISSUE TITLE
NeurIPS '21 | Early Convolutions Help Transformers See Better.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#61]
Xiao T, Singh M, Mintun E, et al. [Early convolutions help transformers see better](https://proceedings.neurips.cc/paper/2021/file/ff1418e8cc993fe8abcfe3ce2003e5c5-Paper.pdf).
Synced. [Facebook & UC Berkeley Substitute a Convolutional Stem to Dramatically Boost Vision Transformers’ Optimization Stability](https://medium.com/syncedreview/facebook-uc-berkeley-substitute-a-convolutional-stem-to-dramatically-boost-vision-transformers-955164b257c2).

**Comment by NorbertZheng** [comment]
## Overview
Recent studies have shown that vision transformer (ViT) models can attain better results than most state-of-the-art convolutional neural networks (CNNs) across various image recognition tasks, and can do so while using considerably fewer computational resources. This has led some researchers to propose ViTs could replace CNNs in this field.

However, despite their promising performance, ViTs are sensitive to the choice of optimizer, the selection of dataset-specific learning hyperparameters, network depth and other factors. CNNs, in contrast, are far more robust and exceptionally easy to optimize.

These ViT/CNN trade-offs inspired a recent study by a research team from Facebook AI and UC Berkeley that proposes a best-of-both-worlds solution. In the paper Early Convolutions Help Transformers See Better, the team employs a standard, lightweight convolutional stem for ViT models that dramatically increases optimizer stability and improves peak performance without sacrificing computational efficiency.

**Comment by NorbertZheng** [comment]
## $ViT_{P}$ v.s. $ViT_{C}$
![image](https://user-images.githubusercontent.com/37235023/217440124-20ca10e8-ae2e-4031-96a6-aad61568795f.png)
ViTs offer the computer vision community a novel alternative to CNNs, and an increasing amount of research is being applied to improve ViTs via
 - multi-scale networks,
 - increased depth,
 - locality priors, etc.

This paper takes a new tact, focusing on ViT models’ unstable optimization issues. The team conjectures that the optimizability differences between ViTs and CNNs lie primarily in the early visual processing performed by ViTs, which “patchifies” the input image into non-overlapping patches to form the transformer encoder’s input set. This ViT patchify stem is implemented as ~~a large-stride convolution~~, whereas studies have shown that
 - in typical CNN designs, best practices converge to a smaller stack of stride-two 3×3 kernels as the network’s stem.

![image](https://user-images.githubusercontent.com/37235023/217440501-c6daf42d-bcbb-409f-8d64-105b18be6635.png)
Based on these factors, the team decided to restrict convolutions in ViT to early visual processing by replacing the patchify stem with its convolutional stem counterpart and removing one transformer block to compensate for the convolutional stem’s extra flops.

**Comment by NorbertZheng** [comment]
## Experiments
To compare the stability of ViT models with the original patchify (P) stem (ViTP) against the convolutional © stem (ViTC), the team conducted experiments on standard ImageNet-1k datasets and reported top-1 error. They chose RegNetY, a state-of-the-art CNN which is easy to optimize, as their good-stability baseline.

![image](https://user-images.githubusercontent.com/37235023/217441188-9ae005d0-4b29-4449-8019-ed65983a1df1.png)
In experiments testing how rapidly networks converge to their asymptotic error, the ViTC converged faster than ViTP on all 50, 100, and 200 epoch schedules. The shortest training schedule (50 epochs) witnessed the most significant improvement: ViTP -1GF had a ten percent top-1 error, while ViTC -1GF reduced this to about six percent.

The researchers also explored how well AdamW and SGD optimize ViT models with the two stem types, with the results indicating that ViTP models suffer a dramatic drop when trained with SGD across all settings, while ViTC models exhibit much smaller error gaps between SGD and AdamW across all training schedules and model complexities.
![image](https://user-images.githubusercontent.com/37235023/217441831-bf4e6ccd-49fb-4703-a05c-73175c97975d.png)
Further, the results from peak performance experiments confirmed that ViTC’s convolutional stem improves not only optimization stability but also model accuracy, delivering ∼1–2 percent top-1 accuracy improvements on ImageNet-1k while maintaining flops and runtime.

**Comment by NorbertZheng** [comment]
## Conclusion
Overall, the study shows that simply replacing the ViT patchify stem with a standard convolutional stem in early visual processing results in marked improvements in terms of optimizer stability and final model accuracy.

**Comment by NorbertZheng** [comment]
## Reference
 - Xiao T, Singh M, Mintun E, et al. [Early convolutions help transformers see better](https://arxiv.org/abs/2106.14881).
